lib/signalfx.js

```diff
@@ -0,0 +1,334 @@
+      body: data,
+      method: 'POST'
+    };
+
+    request(postOptions, function (error, response, body) {
+      _this.asyncRunning = false;
+
+      if (error || response.statusCode !== 200) {
+        winston.info('Filed to sent datapoint: ', response.statusCode, body);
+      }
+    });
+  };
+
+
+// ============================================================================================
+//                  SignalFx API client with Protobuf data sending
+// ============================================================================================
+  /**
+   * SignalFx API client data handler that uses Protocol Buffers.
+   * This class presents the interfaces that handle the serialization of data
+   * using Protocol Buffers
+   *
+   * @param apiToken
+   * @constructor
+   */
+  function ProtoBufSignalFx(apiToken) {
+    SignalFxClient.apply(this, arguments);
+  }
+
+  ProtoBufSignalFx.prototype = Object.create(SignalFxClient.prototype);
+  ProtoBufSignalFx.prototype.constructor = SignalFxClient;
+
+  ProtoBufSignalFx.prototype.getHeaderContentType = function () {
+    return PROTOBUF_HEADER_CONTENT_TYPE;
+  };
+
+  ProtoBufSignalFx.prototype.addToQueue = function (metricType, datapoint) {
+    var protobufDatapoint = new protocolBuffers.DataPoint();
+
+    // Set value by type
+    protobufDatapoint.value = {};
+    if (typeof datapoint.value === 'string') { // string value
+      protobufDatapoint.value.strValue = datapoint.value;
+    } else if (typeof datapoint.value === 'number' && datapoint.value % 1 !== 0) { // 'double' value
+      protobufDatapoint.value.doubleValue = +datapoint.value;
+    } else if (typeof datapoint.value === 'number' && datapoint.value % 1 === 0) { // int value
+      protobufDatapoint.value.intValue = Math.round(+datapoint.value);
+    } else {
+      throw new TypeError('Invalid Value ' + datapoint.value);
+    }
+
+    protobufDatapoint.metricType = protocolBuffers.MetricType[metricType.toUpperCase()];
+    protobufDatapoint.metric = datapoint['metric'];
+
+    if (datapoint['timestamp']) {
+      protobufDatapoint.timestamp = +datapoint['timestamp'];
+    }
+
+    // set datapoint dimensions
+    var dimensions = datapoint['dimensions'] || [];
+    for (var key in dimensions) {
+      if (dimensions.hasOwnProperty(key)) {
+        var dim = {};
+        dim.key = key;
+        dim.value = dimensions[key];
+        protobufDatapoint.dimensions.push(dim);
+      }
+    }
+    this.queue.push(protobufDatapoint);
+  };
+
+  ProtoBufSignalFx.prototype.batchData = function (datapointsList) {
+    var dpum = new protocolBuffers.DataPointUploadMessage();
+    dpum.datapoints = dpum.datapoints.concat(datapointsList);
+
+    try {
+      var dataToSend = dpum.encode();
+    } catch (error) {
+      winston.error('Invalid Protobuf object', error);
+    }
+    return dataToSend.toBuffer();
+  };
+
+
+// ============================================================================================
```

Comment:
Could you please split these different classes to their own files for the sake of organization? Otherwise devs are left to scrolling up and down from one class to another.